rustls/src/crypto/ring/mod.rs

```diff
@@ -29,26 +29,20 @@ pub(crate) mod tls13;
 /// A `CryptoProvider` backed by the [*ring*] crate.
 ///
 /// [*ring*]: https://github.com/briansmith/ring
-pub static RING: &dyn CryptoProvider = &Ring;
+pub fn provider() -> CryptoProvider {
```

Comment:
I wonder if we should give this a more opinionated name? Of course this is not the only possible `CryptoProvider` instantiation based on *ring* cryptography, merely the one with default cipher suites, key exchange groups and signature algorithms. Maybe `default_provider()`?

Reply:
`default_provider` sounds good to me. I agree it would be nice to emphasize the intent is that this is a starting point for sensible defaults that can be further customized.

Reply:
> `default_provider` sounds good to me. I agree it would be nice to emphasize the intent is that this is a starting point for sensible defaults that can be further customized.

I don't object to that. I expect that some packagers of Rustls (e.g. Linux distros) will change the default provider to something that isn't *ring*, like OpenSSL Or Windows CNG. And then people who want/need to use *ring* (e.g. because of FIPS, in the future) will need a way to force the use of *ring*. So ideally there would be both a way to force the use of *ring* and a `default_provider()`.

For the same reason, enabling "FIPS mode" shouldn't affect which provider is used; the option should only change how the provider is configured. So, for example if one wants to use AWS-LC in FIPS mode, they should choose AWS-LC *and* also choose "FIPS mode."

Reply:
I've switched to `default_provider` for the constructors.

>  So ideally there would be both a way to force the use of ring and a default_provider().

That's certainly possible. Using `ClientConfig::builder` or `ServerConfig::builder` will default to *ring*, but if you wanted to make the choice explicit it's just sugar over `builder_with_provider(rustls::crypto::ring::default_provider().into())`.